jolt-core/src/host/mod.rs

```diff
@@ -181,9 +181,9 @@ impl Program {
 
     // TODO(moodlezoup): Make this generic over InstructionSet
     #[tracing::instrument(skip_all, name = "Program::trace")]
-    pub fn trace(mut self) -> (JoltDevice, Vec<JoltTraceStep<RV32I>>) {
+    pub fn trace(&mut self) -> (JoltDevice, Vec<JoltTraceStep<RV32I>>) {
         self.build();
-        let elf = self.elf.unwrap();
+        let elf = self.elf.clone().unwrap();
```

Comment:
we shouldn't need this clone, we pass in `elf` by reference below

Reply:
Self.elf is an `Option<PathBuf>` the clone is required to unwrap it. Previously this API would consume itself `trace(mut self)`, which seemed weird to me so I changed it to `trace(&mut self)` and that means we can't just pull elf out of Self anymore since we just have a ref to Self.

Does that make sense?